File: java/StringUtilities.java
import java.util.Random;
import java.util.UUID;

/**
 * String generation and manipulation utility functions for test automation
 */
public class StringUtilities {

    /**
     * Generate a random string of specified length
     * 
     * @param length Length of the random string
     * @return Random alphanumeric string
     */
    public static String generateRandomString(int length) {
        String characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
        Random random = new Random();
        StringBuilder result = new StringBuilder(length);
        for (int i = 0; i < length; i++) {
            result.append(characters.charAt(random.nextInt(characters.length())));
        }
        return result.toString();
    }

    /**
     * Generate a random integer string of specified length
     * 
     * @param length Length of the random integer string
     * @return Random integer string
     */
    public static String generateRandomInteger(int length) {
        String digits = "0123456789";
        Random random = new Random();
        StringBuilder result = new StringBuilder(length);
        for (int i = 0; i < length; i++) {
            result.append(digits.charAt(random.nextInt(digits.length())));
        }
        return result.toString();
    }

    /**
     * Generate a unique identifier (UUID)
     * 
     * @return UUID string
     */
    public static String generateUUID() {
        return UUID.randomUUID().toString();
    }
}

File: typescript/stringUtilities.ts
/**
 * String generation and manipulation utility functions for test automation
 */

import { v4 as uuidv4 } from 'uuid';

/**
 * Generate a random string of specified length
 * @param length - Length of the random string
 * @returns Random alphanumeric string
 */
export function generateRandomString(length: number = 10): string {
    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
    let result = '';
    for (let i = 0; i < length; i++) {
        result += characters.charAt(Math.floor(Math.random() * characters.length));
    }
    return result;
}

/**
 * Generate a random integer string of specified length
 * @param length - Length of the random integer string
 * @returns Random integer string
 */
export function generateRandomInteger(length: number = 10): string {
    const digits = '0123456789';
    let result = '';
    for (let i = 0; i < length; i++) {
        result += digits.charAt(Math.floor(Math.random() * digits.length));
    }
    return result;
}

/**
 * Generate a unique identifier (UUID)
 * @returns UUID string
 */
export function generateUUID(): string {
    return uuidv4();
}
